httpcore5/src/main/java/org/apache/hc/core5/http/impl/io/HttpRequestExecutor.java

```diff
@@ -212,7 +213,11 @@ public ClassicHttpResponse execute(
             return response;
 
         } catch (final HttpException | IOException | RuntimeException ex) {
-            Closer.closeQuietly(conn);
+            if (http1Config.getUseRstOnTimeout()) {
```

Comment:
Is there a use case for keeping the old behavior?

Reply:
@garydgregory Graceful shutdown of TLS connections would be one. One can drop a TLS connection without a `close-notify` handshake but it is generally considered rude.

@ooeunz Why do not we always use `IMMEDIATE` close for plain (non TLS) conections by default? Would that make the new config flag redundant?

Reply:
To maintain backward compatibility, the existing behavior has been preserved.

However, if it is determined that there are no issues with backward compatibility, I would like to make immediate close the default behavior.

Reply:
TY @ok2c for the explanation. I think details like this should be in the Javadoc for the getter-setter pair.